src/app/static/src/tests/components/plots/barchart/chartjsBar.test.ts

```diff
@@ -0,0 +1,52 @@
+import ChartjsBar from "../../../../app/components/plots/barchart/chartjsBar.vue";
+import {createLocalVue, shallowMount} from "@vue/test-utils";
+
+const localVue = createLocalVue();
+
+describe("chartjsBar component", () => {
+
+    const propsData = {
+        xLabel: "X Axis",
+        yLabel: "Y Axis",
+        chartdata: {
+            labels: ["group1", "group2"],
+            datasets:[
+                {
+                    label: "dataset1",
+                    backgroundColor: "#111111",
+                    data: [1,2],
+                    errorBars: {
+                        "group1": {plus: 1.1, minus: 0.9},
+                        "group2": {plus: 2.1, minus: 1.9}
+                    }
+                }
+            ]
+        }
+    };
+
+    it("renders as expected", () => {
+        const wrapper = shallowMount(ChartjsBar, {propsData, localVue});
+        expect(wrapper.findAll("canvas").length).toEqual(1);
+        console.log(wrapper.html());
+    });
+
+    it("updates render when props chartdata changes", () => {
+        const wrapper = shallowMount(ChartjsBar, {propsData, localVue});
+
+        const mockUpdate = jest.fn();
+        const vm = (wrapper as any).vm;
+        vm.updateRender = mockUpdate;
+
+        const newChartData = {
+            ...propsData.chartdata,
+            labels: ["group1"]
+        };
+        wrapper.setProps({
+            ...propsData,
+            chartdata: newChartData
+        });
+
+        expect(mockUpdate.mock.calls.length).toBe(1);
```

Comment:
Might be good to test that this is called with the right arguments as well (including the axis labels that come from the props)

Reply:
updateRender doesn't have any parameters, but I'll add a test mocking the inherited method (renderChart) and check that gets called with expected data by updateRender